flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/LRUCache.java

```diff
@@ -54,11 +57,25 @@ class LRUCache<K, V> extends LinkedHashMap<K, V> {
 
   @Override
   protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
-    boolean remove = size() > maximumSize;
+    boolean remove = size() > maximumSize && evictEldest;
     if (remove) {
       evictionCallback.accept(eldest);
     }
 
     return remove;
   }
+
+  void haltEviction() {
```

Comment:
I don't like this part of the change. It limits our possibilities of choosing the cache implementation